runtime/worker.rs

```diff
@@ -211,7 +212,21 @@ impl MainWorker {
   ) -> Result<(), AnyError> {
     let id = self.preload_module(module_specifier).await?;
     self.wait_for_inspector_session();
-    self.js_runtime.mod_evaluate(id).await
+    let mut receiver = self.js_runtime.mod_evaluate(id);
+    tokio::select! {
+      maybe_result = receiver.next() => {
+        debug!("received module evaluate {:#?}", maybe_result);
+        let result = maybe_result.expect("Module evaluation result not provided.");
+        return result;
+      }
+
+      event_loop_result = self.run_event_loop() => {
+        event_loop_result?;
+        let maybe_result = receiver.next().await;
+        let result = maybe_result.expect("Module evaluation result not provided.");
+        return result;
```

Comment:
Why does this logic need to be duplicated in JsRuntime::execute_module and in Worker:: execute_module ? Is this handling two different situations?

Reply:
Why doesn't this one handle the has_been_terminated situation?

Reply:
> Why does this logic need to be duplicated in Worker::execute_module and in WebWorker:: execute_module ? Is this handling two different situations?

Yes, as per PR description implementations of event loop are different in `Worker` and `WebWorker`.

Reply:
> Why doesn't this one handle the has_been_terminated situation?

`Worker` has no termination logic, only `WebWorker` does.